numpy/core/src/multiarray/compiled_base.c

```diff
@@ -603,9 +603,18 @@ arr_interp(PyObject *NPY_UNUSED(self), PyObject *args, PyObject *kwdict)
                 dres[i] = dy[j];
             }
             else {
-                const npy_double slope = (slopes != NULL) ? slopes[j] :
-                                         (dy[j+1] - dy[j]) / (dx[j+1] - dx[j]);
+                const npy_double slope =
+                        (slopes != NULL) ? slopes[j] :
+                        (dy[j+1] - dy[j]) / (dx[j+1] - dx[j]);
+
+                /* If we get nan in one direction, try the other */
                 dres[i] = slope*(x_val - dx[j]) + dy[j];
+                if (NPY_UNLIKELY(npy_isnan(dres[i]))) {
+                    dres[i] = slope*(x_val - dx[j+1]) + dy[j+1];
```

Comment:
Looking over it a last time. I think this should be `dx[j+1] - x_val`. It probably never matters, because the only possible way it can normally make a difference is when the slope is 0. But, to be sure about wonky edge-cases...

Reply:
This branch is the one where slope is maybe non-zero (most likely inf). I'm pretty sure this is correct as is. As written, that term will be negative, which is necessary to interpolate the slope backwards rather than forward.

Reply:
Ooops, messed that one up sorry.